module/zfs/zil.c

```diff
@@ -769,11 +766,6 @@ zil_alloc_lwb(zilog_t *zilog, blkptr_t *bp, boolean_t slog, uint64_t txg,
 	list_insert_tail(&zilog->zl_lwb_list, lwb);
 	mutex_exit(&zilog->zl_lock);
 
-	ASSERT(!MUTEX_HELD(&lwb->lwb_vdev_lock));
```

Comment:
are these being removed due to the broader change? I'm just curious.. since we don't initialize these, I think it is nice to verify their contents are as we expect, before using them.. perhaps that's just my personal preference, and not a general rule taken by the project..

Reply:
I don't feel strongly.. when trying to track down bugs, sometimes I just like seeing the state is actually verified/initialized before being used.. just my personal paranoia, I guess.. I'm good either way, you can leave it as-is.

Reply:
> The PR is now based on top of ...

I see now, this isn't even going to land as part of this PR.. sorry for the noise.